The case of missing `specification` property

## This is a...
<!-- Check one of the following options with "x" -->
<pre><code>
[ ] Feature request
[x] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>


## The problem

Running an integration with API connector fails to load Swagger specification. The stack trace is:
```
Caused by: java.lang.IllegalArgumentException: The given Swagger specification could not be loaded from `file:/tmp/swagger-operation1065033880072210483.swagger`. Tried loading using Camel's resource resolution and using Swagger's own resource resolution. Swagger tends to swallow exceptions while parsing, try specifying Java system property `debugParser` (e.g. `-DdebugParser=true`), the exception that occured when loading using Camel's resource loader follows
	at org.apache.camel.component.rest.swagger.RestSwaggerEndpoint.loadSpecificationFrom(RestSwaggerEndpoint.java:471) ~[camel-rest-swagger-2.20.1.jar!/:2.20.1]
	at org.apache.camel.component.rest.swagger.RestSwaggerEndpoint.createProducer(RestSwaggerEndpoint.java:154) ~[camel-rest-swagger-2.20.1.jar!/:2.20.1]
	at org.apache.camel.component.connector.DefaultConnectorEndpoint.createProducer(DefaultConnectorEndpoint.java:51) ~[camel-connector-2.20.1.jar!/:2.20.1]
	at org.apache.camel.impl.ProducerCache.doGetProducer(ProducerCache.java:573) ~[camel-core-2.20.1.jar!/:2.20.1]
	... 107 common frames omitted
Caused by: com.fasterxml.jackson.databind.JsonMappingException: No content to map due to end-of-input
 at [Source: java.io.FileInputStream@e0847a9; line: 1, column: 0]
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:270) ~[jackson-databind-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:3854) ~[jackson-databind-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3799) ~[jackson-databind-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:2337) ~[jackson-databind-2.8.10.jar!/:2.8.10]
	at org.apache.camel.component.rest.swagger.RestSwaggerEndpoint.loadSpecificationFrom(RestSwaggerEndpoint.java:460) ~[camel-rest-swagger-2.20.1.jar!/:2.20.1]
	... 110 common frames omitted
```

Seems that we don't map the properties from the OpenShift secret to the connector at runtime. Case in point the Swagger REST connector (akka custom API connector).

I see the secret was generated as:

```
#
#Tue Mar 06 08:06:44 UTC 2018
swagger-operation.configurations.swagger-operation-3.host=...
sql-1.password=...
swagger-operation.configurations.swagger-operation-3.authenticationType=none
swagger-operation.configurations.swagger-operation-3.specification=...
```

This comes from `configuredProperties` of a Connector and don't seem to be set on the connector.

## Expected behavior

Set `configuredProperties` from the Connector model to the Connector instance.

## Screenshot

![](https://media3.giphy.com/media/l3q2RauzE5Vzf7iYo/giphy.gif)

## Tasks involved / Steps to Reproduce
<!--
Enlist all the acceptance criteria for new features or the steps required to reproduce the bug/regression reported.
-->
1. Create Custom API connector
2. Create an Integration using this connector
3. Observe the exception in the integration log
4.

--- a/app/integration/runtime/src/main/java/io/syndesis/integration/runtime/handlers/EndpointStepHandler.java
+++ b/app/integration/runtime/src/main/java/io/syndesis/integration/runtime/handlers/EndpointStepHandler.java
@@ -20,18 +20,18 @@ import java.net.URISyntaxException;
 import java.util.Map;
 import java.util.Optional;
 
-import io.syndesis.common.util.CollectionsUtils;
-import io.syndesis.common.util.Optionals;
-import io.syndesis.common.util.Predicates;
-import io.syndesis.integration.runtime.IntegrationRouteBuilder;
-import io.syndesis.integration.runtime.IntegrationStepHandler;
-import io.syndesis.common.model.WithConfigurationProperties;
 import io.syndesis.common.model.action.ConnectorAction;
 import io.syndesis.common.model.action.ConnectorDescriptor;
 import io.syndesis.common.model.connection.Connection;
 import io.syndesis.common.model.connection.Connector;
 import io.syndesis.common.model.integration.Step;
 import io.syndesis.common.model.integration.StepKind;
+import io.syndesis.common.util.CollectionsUtils;
+import io.syndesis.common.util.Optionals;
+import io.syndesis.common.util.Predicates;
+import io.syndesis.integration.runtime.IntegrationRouteBuilder;
+import io.syndesis.integration.runtime.IntegrationStepHandler;
+
 import org.apache.camel.model.ProcessorDefinition;
 import org.apache.camel.util.ObjectHelper;
 import org.apache.camel.util.URISupport;
@@ -91,11 +91,7 @@ public class EndpointStepHandler implements IntegrationStepHandler, IntegrationS
         // any configuredProperties on action descriptor are considered
         properties.putAll(descriptor.getConfiguredProperties());
 
-        String uri = componentScheme;
-
-        if (hasComponentProperties(configuredProperties, connector, action)) {
-            uri = String.format("%s-%s", componentScheme, stepIndex);
-        }
+        String uri = String.format("%s-%s", componentScheme, stepIndex);
 
         if (ObjectHelper.isNotEmpty(uri) && ObjectHelper.isNotEmpty(properties)) {
             try {
@@ -114,18 +110,4 @@ public class EndpointStepHandler implements IntegrationStepHandler, IntegrationS
         return Optional.ofNullable(route);
     }
 
-    // *************************
-    // Helpers
-    // *************************
-
-
-    private static boolean hasComponentProperties(Map<String, String> properties, WithConfigurationProperties... withConfigurationProperties) {
-        for (WithConfigurationProperties wcp : withConfigurationProperties) {
-            if (properties.entrySet().stream().anyMatch(wcp::isComponentProperty)) {
-                return true;
-            }
-        }
-
-        return false;
-    }
 }

